lib/crowbar/client/command/proposal.rb

```diff
@@ -101,33 +104,160 @@ module Proposal
+                template = Request.instance.proposal_template(
+                  barclamp
+                ).parsed_response.easy_merge(
+                  "id" => proposal
+                )
+
+                payload = case
+                when opts[:data]
+                  json = begin
+                    JSON.load opts[:data]
+                  rescue
+                    exit_now! "Invalid json data"
+                  end
+
+                  if opts[:merge]
+                    template.easy_merge json
+                  else
+                    json
+                  end
+                when opts[:file]
+                  json = begin
+                    JSON.load opts[:file]
+                  rescue
+                    exit_now! "Invalid json file"
+                  end
+
+                  if opts[:merge]
+                    template.easy_merge json
+                  else
+                    json
+                  end
+                else
+                  begin
+                    editor = Util::Editor.new content: template
+                    editor.edit!
+
+                    editor.result
+                  rescue EditorAbortError => e
+                    exit_now! e.message
+                  rescue EditorStartupError => e
+                    exit_now! e.message
+                  rescue InvalidJsonError => e
+                    exit_now! e.message
+                  rescue
+                    exit_now! "Editing content failed"
+                  end
+                end
+
+                Request.instance.proposal_create(barclamp, proposal, payload) do |request|
+                  case request.code
+                  when 200
+                    say "Created successfully #{proposal} on #{barclamp}"
+                  else
+                    exit_now! request.parsed_response["error"]
+                  end
+                end
               end
             end
 
             parent.desc "Edit a proposal for specific barclamp"
             parent.arg :barclamp
             parent.arg :proposal
-            parent.command :edit do |c|
+            parent.command "edit" do |c|
+              c.desc "Proposal data in json format"
+              c.flag [:data], type: String, default_value: nil
+
+              c.desc "Proposal data as a json file"
+              c.flag [:file], type: String, default_value: nil
+
+              c.desc "Merge input with proposal data"
+              c.switch [:merge], negatable: false
+
               c.action do |global, opts, args|
                 barclamp = args.shift
                 proposal = args.shift
                 helper.validate_availability_of! barclamp
 
-                raise "Not implemented yet! (#{barclamp}, #{proposal})"
+                original = Request.instance.proposal_show(
+                  barclamp,
+                  proposal
+                ).parsed_response
+
+                payload = case
+                when opts[:data]
```

Comment:
Style/CaseIndentation: Indent `when` as deep as `case`. (https://github.com/bbatsov/ruby-style-guide#indent-when-to-case)